subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/DependencyResolveRulesIntegrationTest.groovy

```diff
@@ -66,6 +67,83 @@ class DependencyResolveRulesIntegrationTest extends AbstractIntegrationSpec {
         succeeds("check")
     }
 
+    @Unroll
+    void "project dependency (#versionInProject) vs external dependency (#versionExternal) resolved in favor of #winner, when forcedVersion=#forcedVersion"()
+    {
+        mavenRepo.module("org.utils", "api", versionExternal).publish()
+        if (forcedVersion!=null && forcedVersion!='project' && forcedVersion!=versionExternal) {
+            mavenRepo.module("org.utils", "api", forcedVersion).publish()
+        }
+        mavenRepo.module("org.utils", "api-ext", '1.0').dependsOn("org.utils", "api", versionExternal).publish()
+        settingsFile << 'include "api", "moduleA", "moduleB"'
+
+        String apiDependency = forcedVersion==null && forcedVersion!='project'?
+            'conf project(":api")' : "conf ('org.utils:api:$forcedVersion') { force = true }"
+
+        String substitutionSetting = forcedVersion == 'project'?
+            "dependencySubstitution { substitute module('org.utils:api') with project(':api') }" : ''
```

Comment:
This is never going to be used. We should have a test that demonstrates using dependency substitution to choose an external dependency over a project dependency.